chatbot/middleware/rate_limiting.py

```
@@ -0,0 +1,368 @@
+        
+        # فیلتر کردن درخواست‌های قدیمی
+        recent_requests = [
+            req_time for req_time in requests_data 
+            if req_time > window_start
+        ]
+        
+        # بررسی تعداد درخواست‌ها
+        if len(recent_requests) >= limit_config['requests']:
+            logger.warning(
+                f"Rate limit exceeded for {cache_key}. "
+                f"Requests: {len(recent_requests)}/{limit_config['requests']}"
```

Ruff (0.12.2)

Logging statement uses f-string

(G004)